app/src/androidTest/java/com/hedvig/app/feature/embark/MoreOptionsTest.kt

```diff
@@ -46,4 +66,22 @@ class MoreOptionsTest : TestCase() {
             }
         }
     }
+
+    @Test
+    fun loginButtonShouldOpenLoginMethod() {
+        val newInstance = MoreOptionsActivity.newInstance(context())
+        activityRule.launch(newInstance)
+
+        onScreen<MoreOptionsScreen> {
+            loginButton {
+                click()
+            }
+        }
+
+        onScreen<ZignSecScreen> {
```

Comment:
Avoid starting new activities in a test. Instead, prefer asserting on stubbed intents.

This is the general procedure for doing that:

1. Replace `LazyActivityScenarioRule` with `LazyIntentsActivityScenarioRule`
2. Create a `KIntent` in the `Screen` you're working with
3. Before performing the action that causes the intent to fire, run `yourKIntent { stub() }`
4. After performing the action that causes the intent to fire, assert that it was fired with `yourKIntent { intended() }`

Reply:
Cool, thanks for clarifying! Will change